Source/items.cpp

```diff
@@ -516,50 +514,23 @@ void CalcSelfItems(Player &player)
 
 bool GetItemSpace(Point position, int8_t inum)
 {
-	int xx = 0;
-	int yy = 0;
+	Point xy_candidates[9];
+	int candidate_count = 0;
 	for (int j = position.y - 1; j <= position.y + 1; j++) {
-		xx = 0;
 		for (int i = position.x - 1; i <= position.x + 1; i++) {
-			itemhold[xx][yy] = ItemSpaceOk({ i, j });
-			xx++;
-		}
-		yy++;
-	}
-
-	bool savail = false;
-	for (int j = 0; j < 3; j++) {
-		for (int i = 0; i < 3; i++) { // NOLINT(modernize-loop-convert)
-			if (itemhold[i][j])
-				savail = true;
+			if (ItemSpaceOk({ i, j }))
+				xy_candidates[candidate_count++] = { i, j };
 		}
 	}
 
 	int rs = GenerateRnd(15) + 1;

```

Comment:
It's possible for this to initialise rs to -7. The old code would set the item position to the top left space unconditionally, the new code tries to access out-of-bounds in xy_candidates on platforms with signed modulo arithmetic.

Reply:
Should we wrap it in `abs()`?

Reply:
Calling std::max to clamp to 0 would be better, could even change the limit used by generaternd since it won't impact dungeon generation if someone is willing to re-record the demo